mathics/builtin/lists.py

```diff
@@ -2696,3 +2697,113 @@ def apply(self, expr, levels, evaluation):
             evaluation.message('Reverse', 'ilsmp', Expression('Reverse', expr, levels))
         else:
             return Reverse._reverse(expr, 1, py_levels)
+
+
+class Median(Builtin):
+    """
+    <dl>
+    <dt>'Median[$list$]'
+      <dd>returns the median of $list$.
+    </dl>
+
+    >> Median[{26, 64, 36}]
+     = 36
+
+    For lists with an even number of elements, Median returns the mean of the two middle values:
+    >> Median[{-11, 38, 501, 1183}]
+     = 539 / 2
+
+    Passing a matrix returns the medians of the respective columns:
+    >> Median[{{100, 1, 10, 50}, {-1, 1, -2, 2}}]
+     = {99 / 2, 1, 4, 26}
+    """
+
+    messages = {
+        'rectn': 'Expected a list of numbers or a list of lists at position 1 in ``.'
```

Comment:
The mathematica `Medain::rectn` message is ```Rectangular array of real numbers is expected at position `2` in `1`.```, I think we should say something about requiring the input to be rectangular.